velox/functions/prestosql/types/TimestampWithTimeZoneType.cpp

```diff
@@ -53,15 +60,89 @@ void castFromTimestamp(
   });
 }
 
+void castFromString(
+    const SimpleVector<StringView>& inputVector,
+    exec::EvalCtx& context,
+    const SelectivityVector& rows,
+    VectorPtr& result) {
+  auto timestampWithTzVector = result->asFlatVector<int64_t>();
+  auto rawTimestampWithTzValues =
+      timestampWithTzVector->values()->asMutable<int64_t>();
+  timestampWithTzVector->clearNulls(rows);
+
+  context.applyToSelectedNoThrow(rows, [&](auto row) {
+    if (inputVector.isNullAt(row)) {
+      result->setNull(row, true);
+    } else {
+      auto tsTzStr = std::string_view(
+          inputVector.valueAt(row).data(), inputVector.valueAt(row).size());
+
+      // Find the second space, as that marks the start
+      // of the timezone component.
+      auto pos = tsTzStr.find(" ");
+      pos = tsTzStr.find(" ", pos + 1);
+      auto tsStr = tsTzStr.substr(0, pos);
+      auto ts = util::fromTimestampString(StringView(tsStr));
+      auto tzStr = tsTzStr.substr(pos + 1);
+      auto sessionTzID = util::getTimeZoneID(tzStr);
```

Comment:
Why is this sessionTizId ? This is computed from the input value right ? Different values from different rows in the vector could be of different timezones I think. Did you check with that ?